### PR TITLE
Search monitor mnrva 222

### PR DESCRIPTION
**Pull request by lovedeep92**
**JIRA:**
See ticket here -[]( https://jira.rax.io/browse/MNRVA-222)

**Description:**

1) Added search ability on monitors list page to search for particular monitor.

2) Create new service for searching monitors by passing query parameter.

 Testing:

`npm run test `

**Screenshots:**

(if applicable)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -56,7 +56,7 @@
     "zone.js": "0.10.2"
   },
   "devDependencies": {
-    "@angular-devkit/build-angular": "^0.901.7",
+    "@angular-devkit/build-angular": "^0.901.8",
```
Comment (FarajiA) [inline package.json:59]: We'll want exact versions here.

Comment (lovedeep92) [reply package.json:59]: ohh .. okay .. let me do the changes!

Comment (lovedeep92) [reply package.json]: updated!!

File: src/app/_features/monitors/components/list/monitorslist.component.spec.ts [hunk 2]
```diff
@@ -132,4 +132,12 @@ describe('MonitorslistComponent', () => {
     component.prevPage();
     expect(component.page).toEqual(1);
   });
+
+  it('should reset the search results when click on cross button', () => {
```
Comment (FarajiA) [inline src/app/_features/monitors/components/list/monitorslist.component.spec.ts:136]: Here's an example of how I test the reset search results:
https://github.com/racker/Minerva/blob/master/src/app/_features/resources/components/search/search.component.spec.ts#L60

File: src/app/_features/monitors/components/monitor-search/monitor-search.component.ts [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+  ngAfterViewInit():void {
+    const search$ = fromEvent(this.searchMonitors.nativeElement, 'keyup')
+    .pipe(
+      map((e:any) => {
+        if (e.target.value === "") {
+          this.reset();
+        }
+        return e.target.value }), // retrieve the value of the input
+     /*tap((input) => {
```
Comment (FarajiA) [inline src/app/_features/monitors/components/monitor-search/monitor-search.component.ts:40]: This block can be removed.

File: src/app/_features/monitors/components/list/monitorslist.component.spec.ts (hunk 2 above)
Comment (lovedeep92) [reply src/app/_features/monitors/components/list/monitorslist.component.spec.ts:136]: ohh .. yeah .. removed 👍

File: src/app/_features/monitors/components/monitor-search/monitor-search.component.ts (hunk 3 above)
Comment (lovedeep92) [reply src/app/_features/monitors/components/monitor-search/monitor-search.component.ts:40]: yup .. done 👍

File: src/app/_features/monitors/components/list/monitorslist.component.ts [hunk 4]
```diff
@@ -30,7 +30,7 @@ export class MonitorslistComponent implements OnInit {
           this.spnService.changeLoadingStatus(false);
           this.monitors = this.monitorService.monitors.content;
           this.total = this.monitorService.monitors.totalElements;
-          this.searchPlaceholderText = `Search ${this.total} monitors`;
+          this.monitorSearchPlaceholderText = `Search ${this.total} monitors`;
```
Comment (pras7430) [inline src/app/_features/monitors/components/list/monitorslist.component.ts:33]: I think we can remove duplicate code.
ngAfterViewInit and ngOnInit setting the same values(duplicate code). we make a separate function for that

File: src/app/_services/monitors/monitor.service.ts [hunk 5]
```diff
@@ -179,4 +179,28 @@ export class MonitorService {
         );
       }
   }
+
+  searchMonitors(search:string): Observable<Monitors> {
+    if (environment.mock) {
+      let mocks = Object.assign({}, this.mockedMonitors.collection);
+      this.monitors = mocks;
+      let slicedData = [... mocks.content.slice(0 * 10, 1 * 10)];
```
Comment (pras7430) [inline src/app/_services/monitors/monitor.service.ts:187]: let slicedData = [... mocks.content.slice(0 * 10, 1 * 10)];
vs 
 let slicedData = [... mocks.content.slice(0 ,  10)];

File: src/app/_features/monitors/components/list/monitorslist.component.html [hunk 6]
```diff
@@ -1,22 +1,5 @@
...
-            class="hxClear"
-            aria-label="Clear search"
-            hidden>
-            <hx-icon type="times"></hx-icon>
-          </button>
-          <hx-search></hx-search>
-        </hx-search-control>
-  </div>
-
+  <app-monitor-search class="hxCol" [placeholder]="monitorSearchPlaceholderText" (monitorSearchResults)="monitorResults($event)" (searching)="monitorsSearch($event)" (resetResults)="resetSearch()"></app-monitor-search>
```
Comment (pras7430) [inline src/app/_features/monitors/components/list/monitorslist.component.html:2]: Now for every control, we can have an id, as QE guys working with "xpath" that is not correct.
What you guys think.

**Pull request merged**